### ISSUE TITLE
Comment stream

### ISSUE THREAD

**Issue by cmcarey** [trevorsenior/snoocore#139]
Is there any way to, similar to PRAW, loop over for example new comments, and call a function when a new comment has appeared?  Seems to be pretty basic functionality for a bot, but I can't find it.

**Comment by trevorsenior** [comment]
Currently there isn't a built-in way to do this other than looping over `/new` over and over and diffing the results (which is how the implementation would work if it was built in), however this is a common pattern for bots so I don't mind looking into adding in support. 

The implementation will use [event emitters](https://nodejs.org/api/events.html#events_emitter_removelistener_event_listener) / use that API so you can do stuff like:

```javascript
emiter.on('update', function(results) { /* do something with updated results */ });
```

**Comment by cmcarey** [comment]
Yep, I built a working implementation using the /new stream.  Very basic, no event emitters, simply proof of concept.  Also, instead of diffing over results, I just added a `before` parameter to the request with it set to the fullname id of the first result from the previous request - basically after first run it only returns new comments / posts.  Saves having to diff manually.

Also I noticed that occasionally it would fail out (post would vanish), not sure why that happens, I assume it's because of automod / spam filters or something catching and removing the post, so it can't do before.  To fix I added a fallback to the second first post from the previous request.

Works well, you could feasibly have a 10-15 second delay before you loop again (only a few posts a second) and still be safe.

```
var parm = {limit: 100};
function doLoop() {
	try {
		reddit('/r/all/new').get(parm).then(function(data) {
			try {
				parm.before = data.data.children[0].data.name;
				try {
					parm.safebefore = data.data.children[1].data.name;
				} catch(ex) {

				}
			} catch(ex) {
				parm.before = parm.safebefore;
			}
			var children = data.data.children;
                        //children returns new data
		});
	} catch(ex) {
		throw ex;
	}
	
}
```

Sorry for the poor code.

**Comment by trevorsenior** [comment]
If you have any suggestions on the interface please share! 

I'll be thinking of a clean solution in the meantime and start work on it next week. I'll be away this weekend.

**Comment by denizdogan** [comment]
Any news on this feature?

**Comment by trevorsenior** [comment]
No news @denizdogan. I've lost interest in reddit development and this project is not receiving any major updates authored by me. I'm keeping it on GitHub in hopes that someone can take over development vs. silently pulling it.
